app/code/community/Bolt/Boltpay/Helper/Api.php

```diff
@@ -263,8 +263,25 @@ public function createOrder($reference, $sessionQuoteId = null, $isAjaxRequest =
         $service = Mage::getModel('sales/service_quote', $immutableQuote);
 
         try {
+            ///////////////////////////////////////////////////////
+            /// These values are used in the observer after successful
+            /// order creation
+            ///////////////////////////////////////////////////////
+            Mage::getSingleton('core/session')->setBoltTransaction($transaction);
+            Mage::getSingleton('core/session')->setBoltReference($reference);
+            Mage::getSingleton('core/session')->setWasCreatedByMagento($isAjaxRequest);
```

Comment:
order is always created via magento. It's either via Ajax request or via hook.
Maybe instead `wasCreatedByHook`?

Reply:
yep!